enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,427 @@
+
+IP configuration for a given network device may be derived from three configuration mechanisms:
+1. DHCP
+2. An external IPAM IP Pool
+3. Static IP configuration defined in the provider spec
+
+The machine API `VSphereMachineProviderSpec.Network` will be extended to include a subset of additional properties as defined in https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/apis/v1beta1/types.go.  See [openshift/api#1338](https://github.com/openshift/api/pull/1338) for further details on the API extension to the provider specification.  
+
+
+### Workflow Description
+
+#### Installation
+1. OpenShift administrator reserves IP addresses for installation.
+2. OpenShift administrator constructs `install-config.yaml` to define an network device configuration for each node that will receive a static IP address.
+3. OpenShift administrator initiates an installation with `openshift-install create cluster`.  
+4. The installer will proceed to:
+- provision bootstrap and control plane nodes with the specified network device configuration
+- create machine resources containing specified network device configuration
+5. Once the machine API controllers become active, the compute machine resources will be rendered with the specified network device configuration.
+
+#### Scaling new Nodes without `machinesets`
+1. OpenShift administrator reserves IP addresses for new nodes to be scaled up.
+2. OpenShift administrator constructs machine resource to define a network device configuration for each new node that will receive a static IP address.
+3. OpenShift administrator initiates the creation of new machines by running `oc create -f machine.yaml`.  
+4. The machine API will render the nodes with the specified network device configuration.
+
+#### Scaling new Nodes with `machinesets`
+1. OpenShift administrator configures a machineset with `addressesFromPools` defined in the platform specification.
+2. OpenShift administrator or machine autoscaler scales machines
+3. If `addressesFromPools` contains a `AddressesFromPool` definition the machine controller will create an `IPAddressClaim`. The machine controller will:
+- Set a finalizer on the `IPAddressClaim` called `machines.openshift.io/ip-claim-protection`
+- Set an owner reference on the `IPAddressClaim` to the associated `Machine`.
+- Set the condition `IPAddressClaimed` of the associated `Machine` to indicate that it it awaiting IPAddressClaims to be bound
+- Block the creation of the underlying machine in the infrastructure until all associated `IPAddressClaim`s are bound
+4. An external controller will watch for `IPAddressClaim` instances that reference a address pool type known to the external controller.
+5. The external controller will create an `IPAddress` and bind it to its associated `IPAddressClaim`.  
+6. The machine controller will update the condition `IPAddressClaimed` of the associated `Machine` to indicate its `IPAddressClaim`(s) is bound.
```

Comment:
What will we do if there are multiple defined network interfaces with multiple IPAddressClaims? Options:
* Have multiple `IPAddressClaimed` conditions
* Have a single `IPAddressClaimed` condition which is updated only when all IPAddressClaims are fulfilled.

I suggest the latter is the most useful. Would there be any implications for other logic here?

Bikeshed: Suggest calling the condion `IPAddressesClaimed`.

Also we should aim to keep this condition consistent across all providers.